iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -35,9 +36,9 @@ public class OpenHouseInternalCatalog extends BaseMetastoreCatalog {
 
   @Autowired HouseTableRepository houseTableRepository;
 
-  @Autowired
-  @Qualifier("LegacyFileIO")
-  FileIO fileIO;
+  @Autowired FileIOManager fileIOManager;
```

Comment:
[Comments For discussion, no AI needed]
When you do the delta prototype, is this whole catalog implementation re-done using Delta's counterpart, so that we won't need to instantiate this bean ? If that's the case, it is nice that we separate `FileIOManager` out of `StorageManager` !

Reply:
➕ that's my plan! FileIOManager should only be internal to iceberg layer and StorageManager is OH-level. I explored this idea in [design decision:Can we add a top level TableStorage abstraction for combining all FS operations?]( https://docs.google.com/document/d/1gSQhmn7SVsjVCipojSnlx2-6rRBq1-hK-KiFRuWTmXU/edit#heading=h.pueww4o8c12s). For delta we'd need to reimplement it appropriately.